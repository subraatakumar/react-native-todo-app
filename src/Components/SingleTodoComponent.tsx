import React from 'react';
import {StyleSheet, Text, View, Pressable} from 'react-native';
import {Verify, Trash} from 'iconsax-react-native';
import {TodoObject} from '../../types';

interface Props {
  singleTodo: TodoObject;
  handleDeletePress: (id: number) => void;
}
const SingleTodoComponent: React.FC<Props> = ({
  singleTodo,
  handleDeletePress,
}) => {
  return (
    <View style={[styles.singleTodoContainer, styles.shadowProp]}>
      <Text style={styles.todoText}>{singleTodo.text}</Text>
      <View style={styles.singleTodoIconsContainer}>
        <Pressable
          testID="delete-btn"
          onPress={() => handleDeletePress(singleTodo.id)}>
          <Trash size="45" color="#FF0000" />
        </Pressable>
        <Pressable onPress={() => handleDeletePress(singleTodo.id)}>
          <Verify size="45" color="#21A300" />
        </Pressable>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  bodyStyle: {
    justifyContent: 'flex-end',
  },
  navBar: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 14,
  },
  navBarDate: {
    fontFamily: 'Roboto',
    fontWeight: 'bold',
    fontSize: 24,
    color: '#000000',
  },
  textInputContainer: {
    flexDirection: 'row',
    padding: 14,
  },
  textinput: {
    borderWidth: 1,
    flex: 1,
    borderRadius: 5,
    fontFamily: 'Roboto',
    fontSize: 24,
    paddingLeft: 15,
    paddingRight: 15,
  },
  submitPressable: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#BBECA4',
    borderRadius: 5,
    padding: 10,
    width: 53,
    marginLeft: 15,
  },
  submitArrow: {
    padding: 10,
  },
  singleTodoContainer: {
    margin: 10,
    padding: 15,
    backgroundColor: '#FBFBFB',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: '#FBFBFB',
  },
  singleTodoIconsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  todoText: {
    fontFamily: 'Roboto',
    fontSize: 24,
    color: '#000000',
    paddingBottom: 20,
  },
  shadowProp: {
    shadowColor: '#000000',
    shadowOffset: {width: 4, height: 4},
    shadowRadius: 1,
    elevation: 10,
  },
});

export default SingleTodoComponent;
