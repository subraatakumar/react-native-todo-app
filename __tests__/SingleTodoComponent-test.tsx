import React from 'react';
import renderer from 'react-test-renderer';
import SingleTodoComponent from '../src/Components/SingleTodoComponent';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <SingleTodoComponent
        handleDeletePress={() => {}}
        singleTodo={{id: 5, date: 123456, text: 'task1'}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
