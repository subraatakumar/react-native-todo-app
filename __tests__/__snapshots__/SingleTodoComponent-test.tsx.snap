// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FBFBFB",
        "borderColor": "#FBFBFB",
        "borderRadius": 5,
        "borderWidth": 1,
        "margin": 10,
        "padding": 15,
      },
      Object {
        "elevation": 10,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": 4,
        },
        "shadowRadius": 1,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Roboto",
        "fontSize": 24,
        "paddingBottom": 20,
      }
    }
  >
    task1
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="45"
        bbWidth="45"
        fill="none"
        focusable={false}
        height="45"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 45,
              "width": 45,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M21 5.98c-3.33-.33-6.68-.5-10.02-.5-1.98 0-3.96.1-5.94.3L3 5.98M8.5 4.97l.22-1.31C8.88 2.71 9 2 10.69 2h2.62c1.69 0 1.82.75 1.97 1.67l.22 1.3M18.85 9.14l-.65 10.07C18.09 20.78 18 22 15.21 22H8.79C6 22 5.91 20.78 5.8 19.21L5.15 9.14M10.33 16.5h3.33M9.5 12.5h5"
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={4294901760}
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="1.5"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="45"
        bbWidth="45"
        fill="none"
        focusable={false}
        height="45"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 45,
              "width": 45,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m8.38 12 2.41 2.42 4.83-4.84"
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={4280394496}
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="1.5"
          />
          <RNSVGPath
            d="M10.75 2.45c.69-.59 1.82-.59 2.52 0l1.58 1.36c.3.26.86.47 1.26.47h1.7c1.06 0 1.93.87 1.93 1.93v1.7c0 .39.21.96.47 1.26l1.36 1.58c.59.69.59 1.82 0 2.52l-1.36 1.58c-.26.3-.47.86-.47 1.26v1.7c0 1.06-.87 1.93-1.93 1.93h-1.7c-.39 0-.96.21-1.26.47l-1.58 1.36c-.69.59-1.82.59-2.52 0l-1.58-1.36c-.3-.26-.86-.47-1.26-.47H6.18c-1.06 0-1.93-.87-1.93-1.93V16.1c0-.39-.21-.95-.46-1.25l-1.35-1.59c-.58-.69-.58-1.81 0-2.5l1.35-1.59c.25-.3.46-.86.46-1.25V6.2c0-1.06.87-1.93 1.93-1.93h1.73c.39 0 .96-.21 1.26-.47l1.58-1.35Z"
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={4280394496}
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="1.5"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
